### PR TITLE
Fix static file caching is disabled in Airflow Webserver.

### LINKED ISSUE #39344: webserver static file caching is disabled above 2.3.0 version.
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.3.x ~ 2.9.x

### What happened?

Recently, I migrated airflow version 2.2.0 to 2.9.0 and python 3.8 to 3.12.

and i experienced too much load time in airflow web dashboard.
in airflow 2.2.0, airflow dashboard loaded almost instantly. but in 2.9.0, it tooks almost 10 seconds.

## Airflow 2.2.0
image

## Airflow 2.9.0
image

### What you think should happen instead?

I think there is no reason to make slow page loading time because of not using static file caching.

And I figured out the root cause. the problem is airflow webserver's caching is disabled because of bumping up Flask version 1.x to 2.x.

In Flask, there is Configuration Option for static file's Expire time for cache-control.
it is `SEND_FILE_MAX_AGE_DEFAULT`.

In Flask 1.1.x, the default value of `SEND_FILE_MAX_AGE_DEFAULT` is 12 hours.
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.send_file_max_age_default
image

After Flask 2.0.x, the default value of `SEND_FILE_MAX_AGE_DEFAULT` is changed to None.
https://flask.palletsprojects.com/en/2.0.x/api/#flask.Flask.send_file_max_age_default
image

after Airflow 2.3.0, Flask version is upgraded to 2.0.x.

When i added flask configuration in app.py
static file caching works well.
image
290_with_caching
example is here. https://github.com/VertexToEdge/airflow-docker-compose

### How to reproduce

I followed setup instruction in https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html
1. Setup Airflow 2.9.0 with docker-compose
``` bash
curl -LfO 'https://airflow.apache.org/docs/apache-airflow/2.9.0/docker-compose.yaml'
AIRFLOW_UID=$(id -u) docker compose up
```

2. access airflow webserver with webbrowser. and refresh the page.

### Operating System

Ubuntu 22.04.3 LTS

### Versions of Apache Airflow Providers

_No response_

### Deployment

Docker-Compose

### Deployment details

_No response_

### Anything else?

I am newbie in Airflow.
I read airflow installation guide and configruation reference. but I'm not sure if there is a better way to do this without changing the code.
Is there any way to configure static file caching? please just talk to me.

thanks for reading. have a good day.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by VertexToEdge**
close #39344 

Above Airflow 2.3.0, webserver's static file caching is disabled because of major updated Flask's breaking change. In Flask 2.0.x, default value of `SEND_FILE_MAX_AGE_DEFAULT` is changed from 12 hours to None on that version.
`SEND_FILE_MAX_AGE_DEFAULT` affects `Expires` Field in HTTP response header.
if it is None, `Expires` Field in HTTP response header is removed.

this pr add `SEND_FILE_MAX_AGE_DEFAULT` config for Flask explicitly in `/airflow/www/app.py`.
It enables static file caching in Airflow webserver.

I checked Airflow configuration references and searched `SEND_FILE_MAX_AGE_DEFAULT` in Airflow codebase, but there's no way to enable static file caching for airflow webserver's Flask.

## Airflow 2.9.0: As-Is
image

## Airflow 2.9.0: To-Be (Caching Worked)
290_with_caching

## Local test in /tests/www
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bbovenzi** [comment]
Nice catch! We could probably cache it for a few weeks. Assets will only change between airflow version updates.

File: airflow/www/app.py [hunk 1]
```diff
@@ -120,6 +120,10 @@ def create_app(config=None, testing=False):
         )
         cookie_samesite_config = "Lax"
     flask_app.config["SESSION_COOKIE_SAMESITE"] = cookie_samesite_config
+
```
Comment (Taragolis) [inline airflow/www/app.py]: Could you remove whitespaces in this line.

You could do this manually, or by run `ruff` (if it installed), either:
- `ruff check --fix`
- `pre-commit run ruff --all-files` (if pre-commit installed)

Comment (VertexToEdge) [reply airflow/www/app.py]: It's my mistake... I ran pre-commit.

**Comment by jscheffl** [comment]
Really cool catch. I can immediately see it locally and the UI feels immediately faster with this PR! THANKS!

(I think we could debate about the time, 12h sounds reasonable for me, still it will allow also proxy chains to cache files (if not HTTPS) - if somebody wants to have this nicer, still could be configurable. But 12h is a big improvement anyway!

**Review by jscheffl — APPROVED** [review]

**Pull request merged**